Shell/ConditionalRedundancyHandler.cpp

```diff
@@ -0,0 +1,564 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+/**
+ * @file ConditionalRedundancyHandler.cpp
+ * Implements class ConditionalRedundancyHandler.
```

Comment:
I would be considerably happier if there were a (possibly brief) explanation of what's going on here at an intuitive level.